client/my-sites/upgrades/checkout/domain-details-form.jsx

```diff
@@ -73,7 +81,7 @@ export default React.createClass( {
 		const sanitizedFieldValues = Object.assign( {}, fieldValues );
 		this.fieldNames.forEach( ( fieldName ) => {
 			if ( typeof fieldValues[ fieldName ] === 'string' ) {
-				sanitizedFieldValues[ fieldName ] = fieldValues[ fieldName ].trim();
+				sanitizedFieldValues[ fieldName ] = deburr( fieldValues[ fieldName ].trim() );
```

Comment:
What do you think about call `deburr` on change handler? This way it will not cause a flick / lagged update.

BTW: Is it really required to type in latin letters? That seems a huge oversight for i18n..